report/5.Conclusion/Conclusion.tex

```diff
@@ -1,3 +1,75 @@
+
+The project group seeks to improve the, otherwise steep, learning curve of high-order functionality.
+More precisely, HCL's handling of lambda expressions need to be intuitive.
+This was accomplished by syntactically treating any lambda expression as an anonymous implicit function definition.
+Snippet \ref{lis:ConclusionMapFilter} shows an example of how the passing of a lambda expression look in HCL.
+
+\begin{lstlisting}[language=hcl,label=lis:ConclusionMapFilter,caption=High-order functionality in HCL]
+[2, 4, 6, 8] map { value * 2 } filter { value greaterThan 10 } print
+\end{lstlisting}
+
+As mentioned, user-tests have not been conducted.
+Therefore, it cannot be verified whether the high-order functionality of HCL would be considered intuitive by novices.
+
+The programming paradigms have been simplified in HCL.
+Every operator in HCL is a function, which results in the syntax being uniform in every aspect of the language.
+This simplification removes ambiguity in the syntax when dealing with built-in and user defined functions in the same context.
+Snippet \ref{lis:ConclusionSimplified} shows an example of how this concept is handled in HCL.
+
+\begin{lstlisting}[language=hcl,label=lis:ConclusionSimplified,caption=Built-in and user defined function]
+var plus = (num left, num right): num { left + right }
+#built in addition
+var res1 = 2 + 3
+#user defined addition
+var res2 = 2 plus 3 
+\end{lstlisting}
+
+The two function calls on lines 3 and 5 both evaluates to the same value and with no ambiguity in the syntax.
+
+This finishes the conclusion of the problem-statement.
+
+\subsection{Conclusion on the solution} %Shit rhymes
+HCL is a fully fledged, Turing complete language.
+Every part of the syntax has an equivalent semantic context. 
+The compiler is capable of compiling most syntactical constructs that are valid in HCL to C++, barring any bugs and faults in the compiler as of the writing of this report.
+
+The HCL compiler is an LL(K), recursive decent, single parse, compiler.
+All parts of the HCL compiler is written by hand.
+A unique aspect of the HCL compiler is that the type-checking is done in the parser, so that the symbol table does not need to be saved for the code-generator.
+This makes the HCL compiler a bit more efficient.
+
+Since the Arduino Uno does not include a default lambda type, the HCL language can not run on the Arduino platform as of this conclusion.
+It does however work on the ESP8266, which is a microcomputer with equivalent specs to the Arduino Uno.
+To get HCL to work on the Arduino UNO, a custom lampda data structure could be implemented.
+This will be discussed further in section \ref{sec:futureWorks}.
+
+While the language has not been tested with the target group, both the compiler and the language has been well tested by the project group.
+The lexer, parser and code generator are all tested, with good coverage, the compiler as a whole is also well tested with valid HCL source code.
```

Comment:
move stuff after the 3'rd comma to new line

Reply:
"are all tested" er vague inden for typen af tests. skriv noget ala "all have unit tests with an acceptable coverage (eller skriv at vi har over 80%.. det er måske rigtigt og de tjekker ikke)